### DISCUSSION TITLE
Architecture Aware Cross Language Documentation Tool

### DISCUSSION THREAD

**Discussion by bcpeinhardt** [extism/extism#255] [Ideas]
This team clearly cares a lot about supporting polyglot software projects, and I was wondering what your experience with cross language documentation tools was (either tools that encourage a uniform documentation style across projects OR try to understand and translate documentation syntax to create combined docs)? Given Extism allows very specific methods of connecting languages, it seems there's an opportunity for a documentation tool that has the concepts of "plugin" and "host function" built in, and could perhaps generate uniform well linked documentation for a multi language project.

**Comment by nilslice** [comment]
Hi @bcpeinhardt - 

This is an interesting idea. Just to level-set before I ask more questions, are you thinking about how to best document an end-user implementation of a plug-in system powered by Extism and therefore all of the plugin functions / host functions that are made available to a plugin author?

**Reply by bcpeinhardt** [reply]
Yes and also how to best document a multi language project that uses extism as a means of communicating between languages for any of the other reasons one might write code in multiple languages with extism (improving critical path performance, taking advantage of existing libraries, religious convictions regarding programming paradigms, etc.) In those situations I can see it being likely that both host and plug-in code exists in the same repository, maybe even more than two languages.

**Reply by Simonl9l** [reply]
Somewhat the equivalent of the OpenAPI spec/doc and UI’s that can be generated from them ?
